### PR TITLE
Add containerized jenkins setup

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit adds source files for building a jenkins image with all
the plugins needed/used by the pipeline. This setup can be used to
test the changes to the existing workload or for grabbing the JJB
needed to add a new workload to the pipeline.

### CODE REVIEW

File: containers/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ENV JENKINS_USER admin
+ENV JENKINS_PASS admin
+
+# Skip initial setup
+ENV JAVA_OPTS -Djenkins.install.runSetupWizard=false
+
+COPY plugins.txt /usr/share/jenkins/plugins.txt
+RUN /usr/local/bin/install-plugins.sh < /usr/share/jenkins/plugins.txt
+USER root
+RUN apk add py-pip python3-dev libffi-dev openssl-dev gcc libc-dev make
```
Comment (chaitanyaenr) [inline containers/Dockerfile:12]: Bake in watcher to automate the creation of the jobs.

**Comment by morenod** [comment]
@chaitanyaenr  as we are providing a dockerfile with port, user and password, why no to preselect those values on [config/jjb.ini](https://github.com/cloud-bulldozer/scale-ci-pipeline/blob/master/config/jjb.ini)?

**Review by morenod — APPROVED** [review]

File: containers/plugins.txt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+pipeline-model-definition:1.7.2
+lockable-resources:2.10
+font-awesome-api:5.15.1-1
+popper-api:1.16.0-7
+jquery3-api:3.5.1-2
+bootstrap4-api:4.5.3-1
+echarts-api:4.9.0-2
+okhttp-api:3.14.9
+ansicolor:0.7.3
+metrics:4.0.2.6
```
[the remark this answers is not in the document]
Comment (chaitanyaenr) [reply containers/plugins.txt:118]: Added it, thanks.

**Comment by chaitanyaenr** [comment]
@morenod the production jenkins ( non containerized ) host, user and password might differ depending on the user which is why we didn't set any defaults in config/jjb.ini. The containerized version of the jenkins is to help with the testing the PR's before they go into production jenkins.

**Review by mohit-sheth — APPROVED** [review]
LGTM

**Pull request merged**